### ISSUE TITLE
Tests fail with numpy 2.0

### ISSUE THREAD

**Issue by nsmith-** [cms-nanoAOD/correctionlib#240]
As reported by @lgray
The test that fails is
```
tests/test_core_vect.py:74:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.env/lib/python3.12/site-packages/numpy/_utils/__init__.py:85: in wrapper
    return fun(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<built-in function eq>, array([-99., -99., -99., -99., -99., -99., -99., -99., -99., -99., -99.,
       -99., -99., -...000e+01,  2.69696970e+00, -9.90000000e+01,
       -9.90000000e+01,  2.87878788e+00, -9.90000000e+01, -9.90000000e+01]))
kwds = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError:
E           Arrays are not equal
E
E           Mismatched elements: 33 / 100 (33%)
E           Max absolute difference among violations: 101.87878788
E           Max relative difference among violations: 3266.
E            ACTUAL: array([-99., -99., -99., -99., -99., -99., -99., -99., -99., -99., -99.,
E                  -99., -99., -99., -99., -99., -99., -99., -99., -99., -99., -99.,
E                  -99., -99., -99., -99., -99., -99., -99., -99., -99., -99., -99.,...
E            DESIRED: array([-9.900000e+01, -2.939394e+00, -9.900000e+01, -9.900000e+01,
E                  -2.757576e+00, -9.900000e+01, -9.900000e+01, -2.575758e+00,
E                  -9.900000e+01, -9.900000e+01, -2.393939e+00, -9.900000e+01,...

/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/contextlib.py:81: AssertionError
```

**Comment by nsmith-** [comment]
Can be fixed by upgrading pybind11 to v2.12.0

**Comment by nsmith-** [comment]
I am told by @henryiii that there should have been a warning/error for using pybind11 before v2.12.  Installling correctionlib 2.5 (which uses pybind11 v2.10.0) and numpy 2.0, I don't see any warning/error on executing a test based on the [tutorial example](https://cms-nanoaod.github.io/correctionlib/correctionlib_tutorial.html#Creating-new-corrections):
```python
import correctionlib.schemav2 as cs

ptweight = cs.Correction(
    name="ptweight",
    version=1,
    inputs=[cs.Variable(name="pt", type="real", description="Muon transverse momentum")],
    output=cs.Variable(name="weight", type="real", description="Multiplicative event weight"),
    data=cs.Binning(
        nodetype="binning",
        input="pt",
        edges=[10, 20, 30, 40, 50, 80, 120],
        content=[1.1, 1.08, 1.06, 1.04, 1.02, 1.0],
        flow="clamp",
    ),
)
ptweight.to_evaluator().evaluate([10, 20, 30])
```
and the result of the evaluate call is incorrect:
```
>>> ptweight.to_evaluator().evaluate([10, 20, 30])
array([1.1, 1.1, 1.1])
```

After #242 the issue is fixed
```
>>> ptweight.to_evaluator().evaluate([10, 20, 30])
array([1.1 , 1.08, 1.06])
```
but it is rather disturbing that with numpy v2.0 we silently return incorrect results
